### PR TITLE
remove support for overriding installer image 

### PR DESCRIPTION
**Pull request by staebler**
The support for overriding the installer image has been broken for a while. There is little use for it as one can create a custom release image with whatever installer image is desired.

### CODE REVIEW

**Comment by staebler** [comment]
/hold

This build on #738.

File: pkg/apis/hive/v1/validating-webhooks/clusterimageset_validating_admission_hook.go [hunk 1]
```diff
@@ -148,8 +148,8 @@ func (a *ClusterImageSetValidatingAdmissionHook) validateCreate(admissionSpec *a
 	// Add the new data to the contextLogger
 	contextLogger.Data["object.Name"] = newObject.Name
 
-	if newObject.Spec.InstallerImage == nil && newObject.Spec.ReleaseImage == nil {
-		message := "Failed validation: you must specify either an installer image or a release image"
+	if newObject.Spec.ReleaseImage == "" {
+		message := "Failed validation: you must specify either a release image"
```
Comment (joelddiaz) [inline pkg/apis/hive/v1/validating-webhooks/clusterimageset_validating_admission_hook.go]: "must specify either a release image"
either a release image, or what else are the options? or should this simply be "must specify a release image"?

Comment (staebler) [reply pkg/apis/hive/v1/validating-webhooks/clusterimageset_validating_admission_hook.go]: It should be "must specify a release image": There is no other option.

**Comment by staebler** [comment]
/hold cancel

#738 has merged.

**Pull request merged**